File: JS/src/tests/peekIterator.test.js
const PeekIterator = require('../common/PeekIterator')

const arrayToGenerator = require('../common/arrayToGenerator')

const {assert} = require('chai')

describe('test PeekIterator', () => {
    it('test_peek', () => {
        const it = new PeekIterator(arrayToGenerator([..."abcdefg"]))
        assert.equal(it.next(), 'a')
        assert.equal(it.next(), 'b')
        assert.equal(it.peek(), 'c')
        assert.equal(it.peek(), 'c')
        assert.equal(it.next(), 'c')
        assert.equal(it.next(), 'd')
    })

})

File: Java/test/Common/PeekIteratorTest.java
package Common;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

/**
 * Author : lihuichuan
 * Time   : 2020/11/28
 */
class PeekIteratorTest {

    @Test
    void peek() {
        String source = "abcdefg";
        PeekIterator<Character> it = new PeekIterator<Character>(source.chars().mapToObj(c -> (char) c));

        assertEquals('a', it.next());
        assertEquals('b', it.next());

        it.next();
        it.next();
        assertEquals('e', it.next());
        assertEquals('f', it.peek());
        assertEquals('f', it.peek());

    }

    @Test
    void putBack() {
    }
}